src/Spectre.libClassifier/DatasetFactory.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+* DatasetFactory.cpp
+* Creates OpenCvDataset from external sources.
+*
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <fstream>
+#include <sstream>
+#include "DatasetFactory.h"
+#include "FileOpenException.h"
+
+namespace spectre::supervised
+{
+const int ColumnMatrixWidth = 1;
+
+DatasetFactory::DatasetFactory()
+{
+}
+
+OpenCvDataset DatasetFactory::create(std::string filename) const
```

Comment:
I'm kind of worried that we only check for existence of a text file, but not its validity.

Reply:
yes, you're right, I assume we will have resources in correct format. The validity I think should be somewhere else